doc/source/ray-air/examples/torch_incremental_learning.ipynb

```diff
@@ -0,0 +1,1703 @@
+        "        if train:\n",
+        "            # Only perform random cropping on the Train dataset.\n",
+        "            transform = RandomCrop(28, padding=4)\n",
+        "        else:\n",
+        "            transform = None\n",
+        "        return torchvision.datasets.MNIST(\"./data\", download=True, train=train, transform=transform)\n",
+        "\n",
+        "    def convert_batch_to_pandas(batch):\n",
+        "        images = [np.array(item[0]) for item in batch]\n",
+        "        labels = [item[1] for item in batch]\n",
+        "\n",
+        "        df = pd.DataFrame({\"image\": images, \"label\": labels})\n",
+        "\n",
+        "        return df\n",
+        "\n",
+        "    mnist_dataset = ray.data.read_datasource(\n",
+        "        SimpleTorchDatasource(), dataset_factory=mnist_dataset_factory\n",
+        "    )\n",
+        "    mnist_dataset = mnist_dataset.map_batches(convert_batch_to_pandas)\n",
+        "    return mnist_dataset"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "vqrfgfl9YnVe"
+      },
+      "source": [
+        "## 3b: Create our Stream abstraction\n",
+        "\n",
+        "Now we can create our \"stream\" abstraction. This abstraction provides two\n",
+        "methods (`generate_train_stream` and `generate_test_stream`) that each returns an Iterator\n",
+        "over Ray Datasets. Each item in this iterator contains a unique permutation of\n",
+        "MNIST, and is one task that we want to train on.\n",
+        "\n",
+        "In this example, \"the stream of tasks\" is contrived since all the data for all tasks exist already in an offline setting. For true online continual learning, you would want to implement a custom dataset iterator that reads from some stream datasource to produce new tasks. The only abstraction that's needed is `Iterator[ray.data.Dataset]`.\n",
+        "\n",
+        "Note that the test dataset stream has the same permutations that are used for the training dataset stream. In general for continual learning, it is expected that the data distribution of the test/prediction data follows what the model was trained on. If you notice that the distribution of new prediction queries is changing compared to the distribution of the training data, then you should probably trigger training of a new task."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 5,
+      "metadata": {
+        "id": "f2EagMWCN3he"
+      },
+      "outputs": [],
+      "source": [
+        "from typing import Iterator, List\n",
+        "import random\n",
+        "import numpy as np\n",
+        "\n",
+        "from ray.data import ActorPoolStrategy\n",
+        "\n",
+        "\n",
+        "class PermutedMNISTStream:\n",
+        "    \"\"\"Generates streams of permuted MNIST Datasets.\n",
+        "    \n",
+        "    Args:\n",
+        "        n_tasks: The number of tasks to generate.\n",
```

Comment:
this is really nice. Could you also just add a quick short example usage string? Specifically just using `generate_train_stream` or something

Reply:
and then querying that